### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by MaksTech**
In this pull request:   

- PHP_CodeSniffer was updated to latest [v3.6.0](https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.6.0) version, it supports PHP 8.   
- Also, PHP_CodeSniffer has a couple of new rules regarding naming conventions. Excluded these rules, since they require some changes in all files.
- Fixed a bug that happens on PHP 8

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
> Added composer.lock to Git, because that's what official docs suggest doing.

That makes sense for applications but not for libraries. The main users of the library will not be affected by the lock file in this repo, so it ends up doing more harm than good.

File: src/ArrayComparer/ArrayComparerInterface.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * @license BSD-3-Clause
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-interface ArrayComparer {
+interface ArrayComparerInterface {
```
Comment (JeroenDeDauw) [inline src/ArrayComparer/ArrayComparerInterface.php:16]: Not possible - this is a BIG breaking change. And IMO it is bad practice, explanation at https://verraes.net/2013/09/sensible-interfaces/

Comment (MaksTech) [reply src/ArrayComparer/ArrayComparerInterface.php:16]: Sorry, I was just fixing issues that `phpcs` gave me, based on **your** `phpcs.xml` standard.   
Btw, 2013 is almost a decade ago :)   
https://www.php-fig.org/bylaws/psr-naming-conventions/

**Comment by MaksTech** [comment]
> > Added composer.lock to Git, because that's what official docs suggest doing.
> 
> That makes sense for applications but not for libraries. The main users of the library will not be affected by the lock file in this repo, so it ends up doing more harm than good.

Wondering, what harm would it be? :)   
A quick run through all questions on StackOverflow, the usual answer is "it does not really matter".   
But no problemo, can easily remove it

File: src/ArrayComparer/ArrayComparerInterface.php (hunk 1 above)
Comment (JeroenDeDauw) [reply src/ArrayComparer/ArrayComparerInterface.php:16]: The reasoning is what matters, not the age of the post :) Unfortunate that this got into PSR naming.

File: src/DiffOp/Diff/Diff.php [hunk 2]
```diff
@@ -64,7 +64,7 @@ public function __construct( array $operations = [], $isAssociative = null ) {
 		parent::__construct( [] );
 
 		foreach ( $operations as $offset => $operation ) {
-			if ( !( $operation instanceof DiffOp ) ) {
+			if ( !( $operation instanceof DiffOpInterface) ) {
```
Comment (mariushoch) [inline src/DiffOp/Diff/Diff.php:67]: Missing a space inside the parentheses (after `DiffOpInterface`) and in a few other similar cases (also in other files). Our coding style mandates spaces inside parentheses, for both function calls, like `foobar( 42 );`, and for conditionals like here.

(Unrelated to the PR: It seems that phpcs is not configured to check for this here, although I think it should.)

**Comment by MaksTech** [comment]
- Reverted most changes
- Excluded new naming conventions
- Returned PHP 7.2

**Review by JeroenDeDauw — APPROVED** [review]

**Comment by mariushoch** [comment]
Thank you for tackling this!

**Comment by MaksTech** [comment]
No problem :)   
I want to update the package in the project, how soon can I expect a new version release?

**Pull request merged**